test/core/tsi/ssl_transport_security_test.cc

```diff
@@ -776,10 +776,24 @@ void ssl_tsi_test_handshaker_factory_internals() {
   test_tsi_ssl_client_handshaker_factory_bad_params();
 }
 
+void ssl_tsi_test_duplicate_root_certificates() {
+  const char* root_cert = load_file(SSL_TSI_TEST_CREDENTIALS_DIR, "ca.pem");
```

Comment:
To make sure I understand correctly, you are *not* able to reproduce the bug with this test?

I don't have any particular advice other than debug it I guess? Add some logging every time you call `X509_STORE_add_cert` and make sure it's being called twice. Trace inside the call and see if it's hitting the `X509_R_CERT_ALREADY_IN_HASH_TABLE` case.

Does `root_cert` end in a newline? If not, you may not actually be passing two certificates into your function.

Reply:
Yes, exactly. 

I duplicated the content of root cert and appended it to the root cert file. I also confirmed that `X509_STORE_add_cert` is invoked twice. I probably need to dig further into the Openssl code. 

Yes, it ends in a new line.

Reply:
You can run test with boringssl and with openssl, respectively, and see what happens.